dataactvalidator/scripts/loadTas.py

```diff
@@ -74,23 +74,40 @@ def update_tas_lookups(csv_path):
     new_data = data[data['existing_id'].isnull()]
     del new_data['existing_id']
 
-    # instead of using the pandas to_sql dataframe method like some of the
-    # other domain load processes, iterate through the dataframe rows so we
-    # can load using the orm model (note: toyed with the SQLAlchemy bulk load
-    # options but ultimately decided not to go outside the unit of work for
-    # the sake of a performance gain)
-    for _, row in old_data.iterrows():
-        sess.query(TASLookup).filter_by(account_num=row['account_num']).update(row, synchronize_session=False)
-
-    for _, row in new_data.iterrows():
-        sess.add(TASLookup(**row))
+    if only_fill_in:
+        # Pull in the empty records
+        empty_records = sess.query(TASLookup).filter_by(budget_function_code=None).values('account_num')
+        empty_account_nums = [int(empty_record[0]) for empty_record in empty_records]
+        # find which incoming records can fill in the empty records
+        relevant_old_data = old_data[old_data['account_num'].isin(empty_account_nums)]
+        # Fill them in. If budget_function_code is empty, the following columns have also been empty.
+        fill_in_cols = ['account_title', 'budget_bureau_code', 'budget_bureau_name', 'budget_function_code',
+                        'budget_function_title', 'budget_subfunction_code', 'budget_subfunction_title',
+                        'reporting_agency_aid', 'reporting_agency_name']
```

Comment:
The choice of columns was based all the columns that were completely missing from dev/stg/prod. Can confirm:

```
select * from tas_lookup
where (
	account_title is null and 
	budget_bureau_code is null and
	budget_bureau_name is null and
	budget_function_title is null and
	budget_subfunction_code is null and
	budget_subfunction_title is null and
	reporting_agency_aid is null and
	reporting_agency_name is null)
	and budget_function_code is not null;
```
returns 0 results on dev.  Since the issue arose from not backfilling data when new columns were added, it's safe to say these columns represent the problem.

Reply:
If we wanted a more dynamic solution we could generate this list by figuring out which columns are completely null for future solutions